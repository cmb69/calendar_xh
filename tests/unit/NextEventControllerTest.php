<?php

/**
 * Copyright 2021-2023 Christoph M. Becker
 *
 * This file is part of Calendar_XH.
 *
 * Calendar_XH is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Calendar_XH is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Calendar_XH.  If not, see <http://www.gnu.org/licenses/>.
 */

namespace Calendar;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;

class NextEventControllerTest extends TestCase
{
    public function testIssue51()
    {
        $plugin_tx = XH_includeVar("./languages/en.php", 'plugin_tx');
        $lang = $plugin_tx['calendar'];
        $event = Event::create("1969-03-24", null, "", null, "cmb", "", "", "###");
        $now = LocalDateTime::fromIsoString("2021-03-23T12:34");
        $eventDataService = $this->createStub(EventDataService::class);
        $eventDataService->method("findNextEvent")->willReturn($event);
        $dateTimeFormatter = new DateTimeFormatter($lang);
        $view = new View("./views/", $lang);
        $subject = new NextEventController($lang, $now, $eventDataService, $dateTimeFormatter, $view);
        $response = $subject->defaultAction();
        Approvals::verifyHtml($response);
    }
}
