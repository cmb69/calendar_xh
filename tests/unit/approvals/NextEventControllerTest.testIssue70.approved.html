
<div class="nextevent_date">
  3/24/2022</div>
<div class="calendar_marquee_outer">
  <div class="calendar_marquee">
    <div class="nextevent_event">cmb</div>
    <div class="nextevent_date">52 years</div>
    <div class="nextevent_location">Birthday</div>
  </div>
</div>
